### PR TITLE
Dot product of Effective Hamiltonian Hessian with a change in density matrix

### PR DESCRIPTION
**Pull request by tovrstra**
This is just for testing...

### CODE REVIEW

**Comment by tovrstra** [comment]
It should already be a lot better. I still need to double check a few things though...

**Comment by tovrstra** [comment]
So far, this PR only supports the operation for LDA and GGA restricted calculations. Generalization to MGGA and unrestricted is for later.

**Comment by matt-chan** [comment]
Is this ready to review? We should rebase once the buildkite branch is merged if it is.

**Comment by tovrstra** [comment]
Some more cleaning is needed. Let's first do that and also rebase after buildkite scripts are merged.

**Comment by tovrstra** [comment]
@matt-chan Ready for review.

**Comment by tovrstra** [comment]
@sfias @FarnazH @PaulWAyers This is the last big chunk of old code to be merged into 2.1.0, some of which may be relevant for Chemtools and for computing response properties. Feel free to take a look.

**Comment by matt-chan** [comment]
We really should use typed-memory views in all new code. I'll making an issue for it.

File: horton/meanfield/test/common.py [hunk 1]
```diff
@@ -83,6 +85,232 @@ def check_interpolation(ham, lf, olp, kin, na, exps, do_plot=False):
...
+    exps : Expansion
+        A set of orbitals (one for restricted, two for unrestricted)
+    is_hf : bool
+        Set to True when testing with pure HF.
+    extent : float
+        The extent of the interpolation. 1.0 is whole range.
+    threshold : float
+        The allowed error between the energies.
+    do_plot : bool
+        When True, some plots will be made. Usefull for debugging.
```
Comment (matt-chan) [inline horton/meanfield/test/common.py]: s/usefull/useful/

**Comment by matt-chan** [comment]
@tovrstra Looks good other than the minor fixes. I'll merge after that!

**Comment by matt-chan** [comment]
Looks good. Merging!

**Pull request merged**